src/flake_info.rs

```diff
@@ -135,20 +149,24 @@ impl FlakeMetadata {
     /// this does not ensure anything about the recentness of the locked revs.
     pub async fn check_lock_if_exists(&self) -> Result<()> {
         if self.source_dir.join("flake.lock").exists() {
-            let output = tokio::process::Command::new("nix")
-                .arg("flake")
-                .arg("metadata")
-                .arg("--json")
-                .arg("--no-update-lock-file")
-                .arg(&self.source_dir)
-                .output()
-                .await
-                .wrap_err_with(|| {
-                    eyre!(
-                        "Failed to execute `nix flake metadata --json --no-update-lock-file {}`",
-                        self.source_dir.display()
-                    )
-                })?;
+            let mut command = tokio::process::Command::new("nix");
+            command.arg("flake");
+            command.arg("metadata");
+            command.arg("--json");
+            command.arg("--no-update-lock-file");
+
+            if self.impure {
+                command.arg("--impure");
+            }
+
+            command.arg(&self.source_dir);
+
+            let output = command.output().await.wrap_err_with(|| {
+                eyre!(
+                    "Failed to execute `nix flake metadata --json --no-update-lock-file {}`",
```

Comment:
Shouldn't this error contain `--impure` if used?

Reply:
It should, but didn't notice it could be more dynamic.